src/Microsoft.Performance.SDK/Processing/IProcessingOptionsResolver.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Performance.SDK.Processing.DataSourceGrouping;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    ///     Responsible for choosing the <see cref="ProcessorOptions"/> that will be used for processing.
+    /// </summary>
+    public interface IProcessingOptionsResolver
+    {
+        /// <summary>
+        ///     Get the processor options for a group of data sources processed by a given processor.
+        /// </summary>
+        /// <param name="processingSourceGuid"> Guid for processing source to process the data sources. </param>
+        /// <param name="dataSourceGroup"> A group of data sources. </param>
+        /// <returns> Options for processing.</returns>
```

Comment:
This return value could be much more explicit. Something like
"The `<see cref="ProcessorOptions"/>` to associate with the `<see cref="IProcessingSource"/>` with the given `<paramref name="processingSourceGuid"/>`."